src/metabase/query_processor/middleware/permissions.clj

```diff
@@ -135,7 +135,7 @@
   (when-not (has-data-perms? (required-perms outer-query))
     (check-block-permissions outer-query))
   (when-not (has-data-perms? (query-action-perms outer-query))
-    (throw (perms-exception required-perms))))
```

Comment:
`#function ...` was showing up in the exception; invoking it here makes the actual perms list show up.